### ISSUE TITLE
Get ATLAS non-detections

### ISSUE THREAD

**Issue by genghisken** [genghisken/ps1#48]
Originally reported by: **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**

----------------------------------------

Because ATLAS data is not processed by "skycell" in the way that PS1 data is, there is no simple mechanism to identify non-detections.  Here are my suggestions:

1. Grab all the unique pointing names (quite often just one will appear) then search for all exposures with those pointing names.  The downsides are: a) the pointings are dithered by up to one degree so the object if near the edge will be missed, b) it assumes that pointing names are not changed over time (probably OK).  We could calculate the distance to the centre of the pointings and make a judgement that other pointings may or may not overlap this region.

2. Do a large search for objects around the current one (e.g 1 arcminute).  Grab all unique exposures related to all objects in the search.  Subtract existing unique exposures - and we now have our list of non-detections.  The assumed overlap is only ~ arcmins, so could be OK provided there are other detections in the vicinity.  If there are no detections in the search vicinity we will miss the non-detections.

Both these solutions also assume that ALL data has been ingested from all diff images.  In fact a diff may fail, or the catalogue may contain too many objects (>20,000 at the moment) in which case the data will not be ingested and we miss the non-detection.
 

----------------------------------------
- Bitbucket: https://bitbucket.org/genghisken/ps1/issue/48

**Comment by genghisken** [comment]
*Original comment by* **Ken Smith (Bitbucket: [genghisken](https://bitbucket.org/genghisken), GitHub: [genghisken](https://github.com/genghisken))**:

----------------------------------------

I've added a new search table to the coneSearchHTM code in utils.py which allows me to do a direct search on atlas_diff_detections.  So the algorithm is now:

1. cone search and grab all the atlas_metadata_ids.
2. Remove duplicates (set).
3. Remove atlas_metadata_ids from lightcurve.  The remaining ones are the non-detections.
4. query the MJDs from the atlas_metadata table, given the list of atlas_metadata_ids.

Note for one object, I've experimented with the cone search radius and unique IDs.  Here are the results:

search radius (arcsec) | number of unique exposures | number of detections returned
-----------------------|--------------------------:|-------------------------:
60 | 49 | 85
75 | 56 | 114
150 | 78 | 279
200 | 91 | 480
300 | 98 | 916
400 | 101 | 1423
600 | 102 | 3250
1200 | 111  | 10288

As a compromise it might be wise to choose 150 or at most 200 arcsec.  There's little difference between 200 and 600 arcsec in terms of the number of unique exposures, but a very large difference in the number of objects returned.  So we definitely do NOT want to go as high as 600 arcsec!  The query is extremely expensive (~ 0.5 seconds for 300 arcsec) which is noticeable when clicking on an object.  We may want to inhibit acquisition of non-detections unless the user clicks a check-box.  We could use the larger radius (e.g. 300) for identifying forced photometry exposures which is less time critical.
